SimCalorimetry/HGCalSimAlgos/src/HGCalRawDataPackingTools.cc

```diff
@@ -66,7 +66,7 @@ std::vector<uint32_t> hgcal::econd::produceERxData(const ERxChannelEnable& chann
     if (msb > 0)  // do we have some room for additional information?
       *it_data &= ((1 << msb) - 1);
     if (msb + nbits > 32) {  // spilling onto the next word
-      uint8_t nbits_word1 = 32 - msb - nbits;
+      uint8_t nbits_word1 = 32 - msb;
```

Comment:
Maybe
```suggestion
      uint8_t nbits_word1 = 31 - msb;
```
(because `msb` is bound between 0 and 31, see [L64](https://github.com/cms-sw/cmssw/blob/ff9b49a245a7da8d39c3c3322754173ffe9f4fac/SimCalorimetry/HGCalSimAlgos/src/HGCalRawDataPackingTools.cc#L64))

And perhaps [L68](https://github.com/cms-sw/cmssw/blob/ff9b49a245a7da8d39c3c3322754173ffe9f4fac/SimCalorimetry/HGCalSimAlgos/src/HGCalRawDataPackingTools.cc#L68) should become
```C++
    if (msb + nbits >= 32) {  // spilling onto the next word
```
(but please check, I did not look at the code, I'm just guessing from the lengths of the word)

Reply:
Hi @perrotta, Yes, you are right on both counts. Fixed in https://github.com/cms-sw/cmssw/pull/45665/commits/6cdf0bc42c49bcdcf0ee7f1230059ea5732dde7c Apologies for the noise. I tried running the analyzer myself, but I was too lazy to setup the correct CMSSW release. Hopefully this will fix the warning now.